test/SlowTests/Server/RecordingTransactionOperationsMergerTests.cs

```diff
@@ -0,0 +1,509 @@
+        }
+
+        [Fact]
+        public void RecordingPatchAsBatch()
+        {
+            var filePath = Path.GetTempFileName();
+
+            const string newName = "Balilo";
+            var user = new User { Name = "Baruch" };
+            User[] documents = { };
+
+
+            //Recording
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+                using (var session = store.OpenSession())
+                {
+                    session.Store(user);
+                    session.SaveChanges();
+
+                    session.Advanced.Patch(user, u => u.Name, newName);
+                    session.SaveChanges();
+                }
+                store.Maintenance.Send(new StopTransactionsRecordingOperation());
+            }
+
+            //Replay
+            using (var store = GetDocumentStore())
+            using (var replayStream = new FileStream(filePath, FileMode.Open))
+            {
+                store.Maintenance.Send(new ReplayTransactionsRecordingOperation(replayStream));
+                using (var session = store.OpenSession())
+                {
+                    documents = session.Query<User>().ToArray();
+                }
+                WaitForUserToContinueTheTest(store);
+            }
+
+            //Assert
+            Assert.Equal(documents.First().Name, newName);
+        }
+
+        [Theory]
+        [InlineData("Avi")]
+        [InlineData("Avi", "Baruch", "Charlotte", "Dylan", "Eli", "Fabia", "George", "Harper")]
+        public void RecordingStoreAsBatch(params string[] expected)
+        {
+            var filePath = Path.GetTempFileName();
+            User[] documents = { };
+
+            //Recording
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+                using (var session = store.OpenSession())
+                {
+                    foreach (var name in expected)
+                    {
+                        session.Store(new User { Name = name });
+                    }
+
+                    session.SaveChanges();
+                }
+                store.Maintenance.Send(new StopTransactionsRecordingOperation());
+            }
+
+            //Replay
+            using (var store = GetDocumentStore())
+            using (var replayStream = new FileStream(filePath, FileMode.Open))
+            {
+                store.Maintenance.Send(new ReplayTransactionsRecordingOperation(replayStream));
+                using (var session = store.OpenSession())
+                {
+                    documents = session.Query<User>().ToArray();
+                }
+            }
+
+            //Assert
+            Assert.Contains(documents.Select(u => u.Name), n => expected.Contains(n));
```

Comment:
Move the asserts into the session to check them.
Easier to read

Reply:
done